pkg/controller/elasticsearch/elasticsearch_controller.go

```diff
@@ -295,6 +310,30 @@ func (r *ReconcileElasticsearch) updateStatus(
 	return common.UpdateStatus(r.Client, cluster)
 }
 
+func (r *ReconcileElasticsearch) annotateResource(
+	ctx context.Context,
+	es esv1.Elasticsearch,
+	reconcileState *esreconcile.State,
+) error {
+	span, _ := apm.StartSpan(ctx, "update_annotations", tracing.SpanTypeApp)
+	defer span.End()
+
+	// cluster-uuid is a special case that is not treated here but through an immediate update due to the risk of data loss
+	// see bootstrap package.
+	newAnnotations, err := reconcileState.Annotations()
```

Comment:
nit: using `reconcileState.OrchestrationHints().AsAnnotation()` would help to understand what annotations are set by this part of the code.
```suggestion
	newAnnotations, err := reconcileState.OrchestrationHints().AsAnnotation()
```